### ISSUE TITLE
NS1004 when using Arg.Do()

### ISSUE THREAD

**Issue by rcdailey** [nsubstitute/NSubstitute.Analyzers#159]
I'm honestly not sure if this is a false positive or not. What I can tell you is that my test is behaving as I expect.

Here' the relevant lines:

```cs
            var provider = Substitute.For<IConfigurationProvider<SonarrConfiguration>>();
            var actualActiveConfigs = new List<SonarrConfiguration>();
            provider.ActiveConfiguration = Arg.Do<SonarrConfiguration>(a => actualActiveConfigs.Add(a)); <-- NS1004
```

Using Jetbrains Rider, I get this:
![image](https://user-images.githubusercontent.com/1768054/116010020-7c793c00-a5d1-11eb-841c-2f01b70c2663.png)

My repo is open source so you can easily see the issue in real time here if you want:

https://github.com/rcdailey/trash-updater/blob/7604397/src/Trash.Tests/Config/ConfigurationLoaderTest.cs#L96

I'm not sure why it's getting tripped up. `provider` is itself an interface. It uses a generic parameter that is *not* an interface; maybe that's confusing it?

**Comment by tpodolak** [comment]
Hi @rcdailey, thanks for reporting. Looks like a bug to me. I might have missed the case that Arg.Do can be used with properties. I will try to investigate when I can
